### ISSUE TITLE
Cannot Set Custom XTerm Title

### ISSUE THREAD

**Issue by Marukesu** [elementary/terminal#593]
## What Happened

After #584, terminal don't update the title when printing a xterm escape sequence.

* **Before**
![Terminal Window With The Custom Title](https://user-images.githubusercontent.com/21296444/120875790-08449780-c584-11eb-8225-ba314cff352e.png)

* **After**
![Terminal Window Without The Custom Title](https://user-images.githubusercontent.com/21296444/120875788-067ad400-c584-11eb-80d0-76819f70c405.png)

## Expected Behavior

That the title is updated when a xterm escape sequence is passed.

## Steps to Reproduce

1. Open terminal.
2. Run `echo -n "\033]2;this is a custom title\007"`.
3. Nothing happens.

## Platform Information

elementaryOS 6.0 Early Access

**Comment by jeremypw** [comment]
Not sure that there is a non-hacky way of fixing this while not re-introducing the previous issue. There is no obvious way of determining whether a custom title has been set.  It may be that this is the lesser of two evils.

**Comment by jeremypw** [comment]
I suppose there could be a appmenu option to set a custom title under program control - this could then be protected from the normal updating process.  Whether there is sufficient demand to warrant this is open to question.  Would such a title need to be saved and restored??

**Comment by Marukesu** [comment]
> Not sure that there is a non-hacky way of fixing this while not re-introducing the previous issue. There is no obvious way of determining whether a custom title has been set. It may be that this is the lesser of two evils.

I believe the fix is doing something like that on `.bashrc`:
```bash
case "$TERM" in
xterm*|rxvt*)
    update_title () {
        echo -en "\e]2;${USER}@${HOSTNAME}: ${PWD}\a"
    }

    PROMPT_COMMAND="$PROMPT_COMMAND;update_title"
    ;;
*)
    ;;
esac
```

instead of not listening for it at all. here, that fixed the DConf spam because of `PS1="custom prompt>"`.



> I suppose there could be a appmenu option to set a custom title under program control - this could then be protected from the normal updating process. Whether there is sufficient demand to warrant this is open to question. Would such a title need to be saved and restored??

i don't know if i understand it right, but the escape sequence can be also used by CLI programs to set a custom title while running, i know that at least flatpak-builder do it to show in what step the program is. setting the title via GUI won't fix that case.

**Comment by jeremypw** [comment]
> I believe the fix is doing something like that on .bashrc:

That seems fairly "hacky" to me :wink: 

You're right that a menu option would not help CLI programs.

**Comment by Marukesu** [comment]
> That seems fairly "hacky" to me 

it's just a more complicated version of what is done in `/etc/bash.basrc`:
```bash
# Commented out, don't overwrite xterm -T "title" -n "icontitle" by default.
# If this is an xterm set the title to user@host:dir
#case "$TERM" in
#xterm*|rxvt*)
#    PROMPT_COMMAND='echo -ne "\033]0;${USER}@${HOSTNAME}: ${PWD}\007"'
#    ;;
#*)
#    ;;
#esac
```

however, for some reason, `/etc/skel/.bashrc` use the PS1 for that instead.
```bash
# If this is an xterm set the title to user@host:dir
case "$TERM" in
xterm*|rxvt*)
    PS1="\[\e]0;${debian_chroot:+($debian_chroot)}\u@\h: \w\a\]$PS1"
    ;;
*)
    ;;
esac
```

**Comment by Marukesu** [comment]
I tried to reimplement this by using the `window_title` property if not `""`, otherwise, use "Terminal" for the title and the `current_working_directory` basename for the tabs. it works and don't reintroduce #567 or #583.

![Two Terminals](https://user-images.githubusercontent.com/21296444/123175538-54cf1480-d458-11eb-950b-0e66e751ef93.png)

@jeremypw would a PR be accepted?

**Comment by jeremypw** [comment]
I would certainly be willing to review any PR you produced.

**Comment by jeremypw** [comment]
Re-opening as the fix seems no longer to work.
